### ISSUE TITLE
Implement Encode/Decode for &'a ]

### ISSUE THREAD

**Issue by DrSloth** [SoftbearStudios/bitcode#23]

**Comment by caibear** [comment]
It's a bit tricky since u8s may be packed as u1 u2 u4 which can't be borrowed as u8 from input. It would require borrowing from a bitcode::Buffer after the u8s are unpacked.

**Comment by DrSloth** [comment]
Good point, would it be possible to provide some wrapper type that gets serialized as raw bytes? My specific use case actually is that of non UTF-8 text. I kind of need `str` encoding but not UTF-8.

In my case i could also implement the traits by hand but that doesn't seem to be documented.

**Comment by finnbear** [comment]
And your use-case requires zero-copy for some reason? Otherwise you can use `Vec<u8>` or `Box<[u8]>`.

**Comment by myrgy** [comment]
Hi guys,

Thank you for great framework! Would be nice to see that feature!

Thank you!
